src/mmap_bigarray.ml

```diff
@@ -0,0 +1,85 @@
+module V1 = struct
+  let errno_codes = Unix.[
+    EACCES;
+    EAGAIN;
+    EBADF;
+    EINVAL;
+    ENFILE;
+    ENODEV;
+    ENOMEM;
+    EPERM;
+    EOVERFLOW;
+
+    E2BIG;
+    EBUSY;
+    ECHILD;
+    EDEADLK;
+    EDOM;
+    EEXIST;
+    EFAULT;
+    EFBIG;
+    EINTR;
+    EIO;
+    EISDIR;
+    EMFILE;
+    EMLINK;
+    ENAMETOOLONG;
+    ENOENT;
+    ENOEXEC;
+    ENOLCK;
+    ENOSPC;
+    ENOSYS;
+    ENOTDIR;
+    ENOTEMPTY;
+    ENOTTY;
+    ENXIO;
+    EPIPE;
+    ERANGE;
+    EROFS;
+    ESPIPE;
+    ESRCH;
+    EXDEV;
+    EWOULDBLOCK;
+    EINPROGRESS;
+    EALREADY;
+    ENOTSOCK;
+    EDESTADDRREQ;
+    EMSGSIZE;
+    EPROTOTYPE;
+    ENOPROTOOPT;
+    EPROTONOSUPPORT;
+    ESOCKTNOSUPPORT;
+    EOPNOTSUPP;
+    EPFNOSUPPORT;
+    EAFNOSUPPORT;
+    EADDRINUSE;
+    EADDRNOTAVAIL;
+    ENETDOWN;
+    ENETUNREACH;
+    ENETRESET;
+    ECONNABORTED;
+    ECONNRESET;
+    ENOBUFS;
+    EISCONN;
+    ENOTCONN;
+    ESHUTDOWN;
+    ETOOMANYREFS;
+    ETIMEDOUT;
+    ECONNREFUSED;
+    EHOSTDOWN;
+    EHOSTUNREACH;
+    ELOOP;
+  ]
+
+  let map_file fd ?pos kind layout shared dims =
+    try Bigarray.Genarray.map_file fd ?pos kind layout shared dims
+    with Sys_error description as exn ->
+      try
+        let error_code =
+          List.find
+            (fun errno -> Unix.error_message errno = description) errno_codes
+        in
+        raise (Unix.Unix_error (error_code, "Unix.map_file", ""))
+      with Not_found ->
+        raise exn
```

Comment:
Maybe we should use `raise_with_backtrace`?

Reply:
AFAIK `raise_with_backtrace` requires 4.05.

Reply:
Indeed. I suggest to use the trick I mentioned in the main thread

Reply:
I can do it, but it seems not worth the benefit of having an improved backtrace on only one old compiler version. We select a different implementation for >= 4.06, and `raise_with_backtrace` will be a shim for `raise` on < 4.05.

Reply:
...especially considering how minimal the difference between the stack traces will be.

Reply:
Alright, I accept that